src/dc_mimefactory.rs

```diff
@@ -36,7 +36,7 @@ pub struct dc_mimefactory_t<'a> {
     pub timestamp: i64,
     pub rfc724_mid: *mut libc::c_char,
     pub loaded: dc_mimefactory_loaded_t,
-    pub msg: *mut dc_msg_t<'a>,
+    pub msg: dc_msg_t<'a>,
```

Comment:
@KAction  Since tests segfault now, I suggest a safer way to refactor this: change `msg` type to `Option<dc_msg_t>`, so you can initialize it to `None` and retain all the `is_null` checks.